File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.java
class Solution {
    public int find(int[]nums,int x){ // to find numbers greater than or equal to x
        int count = 0;
        for(int i:nums)
        {
            if(i>=x)
                count++;
        }
      return count;
    }
   
    public int specialArray(int[] nums) {
        int start = 1,end = nums.length;
        
        while(start<=end){
            int x = start + (end-start)/2;
			
             if(x == find(nums,x)) 
                 return x;
            
             if(find(nums,x) > x)
                 start  = x+1;
              else
                  end = x-1;
        }
        
    return -1;
    }
}

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cs
public class Solution {
    public int SpecialArray(int[] nums) {
        Array.Sort(nums);
        for(int i=0; i<nums.Length; i++){
            int n = nums.Length-i;
            bool cond1 = n<=nums[i];
            bool cond2 = (i-1<0) || (n>nums[i-1]);
            if(cond1 && cond2) return n;
        }
        return -1;
    }
}